core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/Rule.scala

```diff
@@ -127,8 +124,11 @@ object Rule {
       }
     }
 
+    protected def caseMappingEquality: UserIdCaseMappingEquality
```

Comment:
IMO at AuthenticationRule level we should have an info what about user ID case comparison strategy. Now, we have how the strategy is working. 

So here we should have instance of this trait:
```
 sealed trait UsernameCaseMapping
  object UsernameCaseMapping {
    case object CaseSensitive extends UsernameCaseMapping
    case object CaseInsensitive extends UsernameCaseMapping
  }
  ```

And when we create a matcher, the data above should be use to create proper function which knows how to apply the strategy during user id comparison.

Reply:
Is `UsernameCaseMapping` `AuthenticationRule` parameter, or attribute? I guess you'd like to close `UsernameCaseMapping` in `AuthenticationRule`. `UsernameCaseMapping` isn't rule attribute, and shouldn't be interpreted bu rule. IMHO ROR should have one instance of ``UsernameCaseMapping`, and it shouldn't be interpreted by any of ror's components itself.